### DISCUSSION TITLE
Extend a configuration

### DISCUSSION THREAD

**Discussion by CavalcanteLeo** [formkit/formkit#756] [Ideas]
Would be awesome if we could extend our configuration with some other configurations, something like eslint, escondfig, tscofig, tailwind and few other does:

```ts
import { anotherConfig } from 'another-project'

export default <DefaultConfigOptions>{
  extends: [anotherConfig] // <= here is the trick
  config: {
    classes: generateClasses(theme),
  },
  icons: {},
  plugins: [],
}
```

**Comment by fenilli** [comment]
if you have another project that has a exported config you can just import and spread it, no need for a new property or anything, its just an object.

**Comment by CavalcanteLeo** [comment]
simply using spread operations doesn't work, but we can easily achieve the object merges with libraries such as defu (https://www.npmjs.com/package/defu)

My case is because we have a monorepo... which contains UI project, project A, project B, project C, and all projects extend from UI, with few different configs, but not completely different

It's working perfectly using defu, but just wish that we could not add a new dependency just for that.

But that's all right
